Change button_widget class to btn-primary

JulienMaille opened the pull request:
FIX #29114

| Q             | A
| ------------- | ---
| Branch?       | master for features / 2.8 up to 4.1 for bug fixes <!-- see below -->
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no     <!-- see https://symfony.com/bc -->
| Deprecations? | no <!-- don't forget to update UPGRADE-*.md and src/**/CHANGELOG.md files -->
| Tests pass?   | yes    <!-- please add some, will be required by reviewers -->
| Fixed tickets | #29114   <!-- #-prefixed issue number(s), if any -->
| License       | MIT
| Doc PR        | symfony/symfony-docs#... <!-- required for new features -->

Change button_widget class to btn-primary in Bootstrap 4 twig template

nicolas-grekas committed:
[HttpFoundation] replace any preexisting Content-Type headers

lyrixx committed:
[HttpFoundation] Fixed PHP doc of ParameterBag::getBoolean

Since the method should return a bool, then the default value should be
a bool too

nicolas-grekas committed:
minor #29111 [HttpFoundation] Fixed PHP doc of ParameterBag::getBoolean (lyrixx)

This PR was merged into the 2.8 branch.

Discussion
----------

[HttpFoundation] Fixed PHP doc of ParameterBag::getBoolean

| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        |

---

Since the method should return a bool, then the default value should be
a bool too

Commits
-------

1fa5a9a0bd [HttpFoundation] Fixed PHP doc of ParameterBag::getBoolean

Nefast committed:
Change button_widget class to btn-primary

javiereguiluz commented on src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig:
```diff
@@ -145,7 +145,7 @@
 
 {% block button_widget -%}
-    {%- set attr = attr|merge({class: (attr.class|default('btn-secondary') ~ ' btn')|trim}) -%}
+    {%- set attr = attr|merge({class: (attr.class|default('btn-primary') ~ ' btn')|trim}) -%}
```
I agree. Let's make this change ... but only for the `submit` button. @xabbuh can you give us any clue about how to do that only for this specific type of button? Thanks!

arkste replied:
@javiereguiluz 

you can either check `type == 'submit'` in `button_widget` or overwrite `class` in a `submit_widget` block.

something like ...

```
{%- block submit_widget -%}
    {%- set attr = attr|merge({class: (attr.class|default('btn-primary') ~ ' btn')|trim}) -%}
    {{- parent() -}}
{%- endblock submit_widget -%}
```

... should work, if you revert the change in `button_widget`.

neFAST replied:
> I guess this should probably not be set for every button but only for submit buttons, shouldn't it?

Well, why would 'secondary' be the new default class for other buttons?

neFAST replied:
I tried and can confirm it works as intended. Should I close that PR and create a new one?

neFAST replied:
@arkste Should we use
`{%- endblock submit_widget -%}`
or
`{%- endblock submit_widget %}`
most of the blocks end without dash (but not all of them)

stof replied:
trimming whitespaces after the `{% endblock %}` tag is useless, as the form theme won't render the spaces being between top-level blocks (as it renders blocks)

xabbuh replied:
@neFAST You can just push additional commits to the branch this PR is based on to update it.

neFAST replied:
@xabbuh Done

xabbuh reviewed (dismissed):
As a bugfix for 3.4?

xabbuh commented:
@neFAST Can you take a look at the failing tests?

neFAST commented:
@xabbuh fixed

xabbuh commented:
@neFAST Thanks for your fast reaction. Nice contribution. 👍 IMO this is a bugfix so I have updated the description accordingly. Don't worry about the target branch. We can rebase your changes on the right branch while merging.

nicolas-grekas committed:
[Dotenv] dont use getenv() to read SYMFONY_DOTENV_VARS

nicolas-grekas committed:
bug #29057 [HttpFoundation] replace any preexisting Content-Type headers (nicolas-grekas)

This PR was merged into the 2.8 branch.

Discussion
----------

[HttpFoundation] replace any preexisting Content-Type headers

| Q             | A
| ------------- | ---
| Branch?       | 2.8
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #21204
| License       | MIT
| Doc PR        | -

Commits
-------

de2ce58a4f [HttpFoundation] replace any preexisting Content-Type headers

nicolas-grekas committed:
Merge branch '2.8' into 3.4

* 2.8:
  [HttpFoundation] Fixed PHP doc of ParameterBag::getBoolean
  [HttpFoundation] replace any preexisting Content-Type headers

nicolas-grekas committed:
bug #29131 [Dotenv] dont use getenv() to read SYMFONY_DOTENV_VARS (nicolas-grekas)

This PR was merged into the 3.4 branch.

Discussion
----------

[Dotenv] dont use getenv() to read SYMFONY_DOTENV_VARS

| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Since using it is not thread safe, let's not rely on it (already done in other places.)

Commits
-------

21a909a189 [Dotenv] dont use getenv() to read SYMFONY_DOTENV_VARS

nicolas-grekas commented:
Thank you @neFAST.
